src/m1_facilitator/lib/TransactionExecutor.ts

```diff
@@ -127,22 +127,30 @@ export default class TransactionExecutor {
   private async execute(): Promise<void> {
     if (!this.mutex.isLocked()) {
       const release = await this.mutex.acquire();
+      Logger.debug('TransactionExecutor:: transaction executor invoked.');
       const transaction = await this.transactionRepository.dequeue();
       try {
-        const nonce = await this.avatarAccount.getNonce(this.web3);
         if (transaction) {
+          Logger.debug('TransactionExecutor:: Executing transaction');
+          const nonce = await this.avatarAccount.getNonce(this.web3);
+          Logger.debug(`TransactionExecutor::Executing transaction ${transaction.id && transaction.id.toString(10)}`);
           const response = await this.sendTransaction(transaction, nonce);
           transaction.transactionHash = response.transactionHash;
           transaction.gas = new BigNumber(response.gas);
           transaction.nonce = nonce;
           await this.transactionRepository.save(transaction);
+          Logger.debug(`TransactionExecutor::Saving transaction ${transaction.id && transaction.id.toString(10)}`);
         }
       } catch (error) {
-        Logger.error(`TransactionExecutor::Error in executing transaction: ${transaction}.
+        Logger.debug('TransactionExecutor::Decreasing nonce in case of error');
+        this.avatarAccount.adjustNonce();
```

Comment:
Can we explicitly name it `decreaseNonce`?

Reply:
Done